src/test/java/pathfinding/ConcreteGraphTest.java

```diff
@@ -74,6 +69,44 @@ public void findPath() throws Exception
 		ArrayList<Node> orderedSolution = graph.findPath(straightNodes[0], straightNodes[straightNodes.length - 1]);
 		assertNotNull(orderedSolution);
 		for (int i = 0; i < 100; i++)
-			assertEquals(orderedSolution.get(i), straightNodes[99-i]);
+			assertEquals(orderedSolution.get(i), straightNodes[99 - i]);
+	}
+
+	@Test
+	public void textDirect() {
+		ConcreteNode oneN = new ConcreteNode();
+		oneN.setX(50);
+		oneN.setY(50);
+		ConcreteNode twoN = new ConcreteNode();
+		twoN.setX(60);
+		twoN.setY(50);
+		ConcreteNode threeN = new ConcreteNode();
+		threeN.setX(60);
+		threeN.setY(40);
+		ConcreteNode fourN = new ConcreteNode();
+		fourN.setX(70);
+		fourN.setY(40);
+		ConcreteNode fiveN = new ConcreteNode();
+		fiveN.setX(70);
+		fiveN.setY(60);
+		ConcreteNode sixN = new ConcreteNode();
+		sixN.setX(75);
+		sixN.setY(60);
+		oneN.addNeighbor(twoN);
+		twoN.addNeighbor(oneN);
+		twoN.addNeighbor(threeN);
+		threeN.addNeighbor(fourN);
+		threeN.addNeighbor(twoN);
+		fourN.addNeighbor(fiveN);
+		fourN.addNeighbor(threeN);
+		fiveN.addNeighbor(sixN);
+		fiveN.addNeighbor(fourN);
+		sixN.addNeighbor(fiveN);
+		ConcreteGraph test = new ConcreteGraph();
+		ArrayList<String> directions = test.textDirect(oneN, sixN, 5);
+		assertEquals("Walk 50 feet", directions.get(1));
+		assertEquals("Walk 100 feet", directions.get(7));
+		assertEquals("Turn left, then", directions.get(6));
```

Comment:
What about the rest of the directions? This array list of directions seems to have 8 different elements, but only 4 are tested. What's supposed to be in those?